### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by stefanpenner**
- remove typeOf vs typeof ambiguity (opting for explicit)
- add `QueryCache.protototype.has` to avoid private usage
- make dummy example work
- add tests to ensure dummy example does not regress.

### CODE REVIEW

File: addon/utils/query-cache.js [hunk 1]
```diff
@@ -22,6 +22,10 @@ export default class QueryCache {
     return this._internalCache[this._getKey(type, key)];
   }
 
+  has(type, key) {
```
Comment (scalvert) [inline addon/utils/query-cache.js:25]: Should we call this `contains`? This seems to be fairly conventional for a cache API.

**Review by scalvert — CHANGES_REQUESTED** [review]

File: addon/utils/query-cache.js (hunk 1 above)
Comment (stefanpenner) [reply addon/utils/query-cache.js:25]: I was mimicing `Map.prototype.has` etc

Comment (scalvert) [reply addon/utils/query-cache.js:25]: ¯\_(ツ)_/¯

**Review by scalvert — APPROVED** [review]

File: addon/utils/query-cache.js (hunk 1 above)
Comment (stefanpenner) [reply addon/utils/query-cache.js:25]: i can gladly swap, your call.

**Pull request merged**